core/domain/state_domain.py

```diff
@@ -1799,6 +1815,19 @@ def update_written_translations(self, written_translations):
         """
         self.written_translations = written_translations
 
+    def update_solicit_answer_details(self, solicit_answer_details):
+        """Update the solicit_answer_details of a state.
+
+        Args:
+            solicit_answer_details: bool. The new value of
+                solicit_answer_details for the state.
+        """
+        if not isinstance(solicit_answer_details, bool):
```

Comment:
We have already done this check in the service layer, isn't it?

Reply:
Yeah this is intended, should I remove this?